postmaster/apiv1/admins.py

```diff
@@ -147,3 +149,112 @@ def unlock_admin(admin_id):
     admin = Admins.query.get_or_404(admin_id)
     clear_lockout_fields_on_user(admin.username)
     return {}, 200
+
+
+@apiv1.route('/admins/<int:admin_id>/2factor', methods=['GET'])
+@login_required
+def twofactor_status(admin_id):
+    """ Returns if 2 factor is enabled or not
+
+        This information is in the main user route,
+        but I added it here as a stub for the URI.
+    """
+    admin = Admins.query.get_or_404(admin_id)
+    return jsonify({"enabled": admin.otp_active})
+
+
+@apiv1.route('/admins/<int:admin_id>/2factor', methods=['PUT'])
+@login_required
+def twofactor_disable(admin_id):
+    """ Disable 2 factor using API.
+
+        Enabling 2 factor from this route is not possible.
+    """
+    admin = Admins.query.get_or_404(admin_id)
+    status = request.get_json(force=True).get('enabled')
+    if status:
+        if status.lower() == "false":
+            admin.otp_active = False
+            try:
+                db.session.add(admin)
+                db.session.commit()
+            except ValidationError as e:
+                raise e
+            except Exception as e:
+                db.session.rollback()
+                json_logger(
+                    'error', current_user.username,
+                    'The following error occurred in twofactor_disable: {0}'.format(str(e)))
+                raise GenericError('The administrator could not be updated')
+            return jsonify({"enabled": admin.otp_active})
+        elif status.lower() == "true":
+            raise GenericError("Cannot enable 2 factor from this route - see docs")
+    raise GenericError("Must provide 'enabled=False'")
+
+
+@apiv1.route('/admins/<int:admin_id>/2factor/qrcode', methods=['GET'])
+@login_required
+def qrcode(admin_id):
+    """ Presents the user with a QR code to scan to setup 2 factor authentication
+    """
+    # render qrcode for FreeTOTP
+    admin = Admins.query.get_or_404(admin_id)
+    if admin.id != current_user.id:
+        raise GenericError('You can only view your user\'s QR code')
+    if admin.otp_active:
+        return "Already gucci"
+    admin.generate_otp_secret()
+    try:
+        db.session.add(admin)
```

Comment:
This line isn't needed because admin is already associated with the session, but it is harmless to have it here.